app/controllers/concerns/paginatable.rb

```diff
@@ -16,6 +16,12 @@ def page_from_params(params)
         page = {}
       end
 
+      # All cursors will need to be decoded from the param
+      if page[:cursor].present?
+        # When we decode and split, we'll always end up with an array
+        page[:cursor] = Base64.strict_decode64(page[:cursor]).split(",")
```

Comment:
Fair enough, this is in part originally why I had this just in each of the controllers I'm still not sure it shouldn't be there. Because arguably it should always be a base64 encoded string if it's coming in from the query string
I guess we can do some safety checks though to ensure we catch potential problems.

Reply:
I would add a `begin/rescue` block, catching an error if the cursor is not properly base64 encoded. The error you need to catch is `ArgumentError` (with message "invalid base64").